CMakeLists.txt

```diff
@@ -189,6 +194,18 @@ target_include_directories(SDL2
     "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>"
 )
 target_compile_definitions(SDL2 PRIVATE "SDL2COMPAT_REVISION=\"${SDL_REVISION}\"")
+if(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+  # Due to a limitation of Microsoft's LTO implementation, LTO must be disabled for memcpy and memset.
+  # The same applies to various functions normally belonging in the C library (for x86 architecture).
+  set_property(SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/SDL_mslibc.c" APPEND_STRING PROPERTY COMPILE_FLAGS " /GL-")
+endif()
+if(WIN32)
+  if(MSVC)
+    set_property(TARGET SDL2 APPEND_STRING PROPERTY LINK_FLAGS " /ENTRY:sdl2_compat_entry")
+  else()
+    set_property(TARGET SDL2 APPEND_STRING PROPERTY LINK_FLAGS " -Wl,--entry=sdl2_compat_entry")
+  endif()
+endif()
```

Comment:
I removed `-nostdlib` again, in favor for explicitly setting the entry function.
This fixes the following link error with msys2's clang32:
```
ld.lld: error: undefined symbol: ___udivdi3
>>> referenced by D:/a/sdl2-compat/sdl2-compat/src/sdl2_compat.c:1338
>>>               CMakeFiles/SDL2.dir/src/sdl2_compat.c.obj:(_Event3to2)
>>> referenced by D:/a/sdl2-compat/sdl2-compat/src/sdl2_compat.c:1381
>>>               CMakeFiles/SDL2.dir/src/sdl2_compat.c.obj:(_Event3to2)
>>> referenced by D:/a/sdl2-compat/sdl2-compat/src/sdl2_compat.c:1384
>>>               CMakeFiles/SDL2.dir/src/sdl2_compat.c.obj:(_Event3to2)

>>> referenced 2 more times
```

Reply:
_DllCRTMain clash is not the reason for -nostdlib: We just do not want any libc / crt calls in the dll. ___udivdi3 etc should  be resolved by linking to clang's rtl (libclang_rt or whatever) just like with gcc linking to libgcc.

Reply:
More importantly the generated dll is broken: note the linker warning:
```
D:/a/_temp/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.1.0/../../../../i686-w64-mingw32/bin/ld.exe: warning: cannot find entry symbol sdl2_compat_entry; not setting start address
```
Because for x86 (and _possibly_ for other 32 bit windows such as arm32), the symbol is prefixed with an underscore.

I'd say not do this at all.

Reply:
How do you explicitly ink to clang_rt?
The library has an arch dependent suffix: https://packages.msys2.org/package/mingw-w64-clang-i686-compiler-rt

Reply:
Don't know. And, honestly didn't know that it'd be that painful.

Reply:
First hits in [Google search](https://www.google.com/search?q=explicitly+link+to+clang_rt) shows that it _is pain_:
https://discourse.llvm.org/t/improve-autolinking-of-compiler-rt-and-libc-on-windows-with-lld-link/71392